python/cuml/test/test_lobpcg.py

```diff
@@ -0,0 +1,177 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cupy as cp
+
+import pytest
+
+from cuml.sparse.linalg.lobpcg import lobpcg as cp_lobpcg
+
+# from numpy.testing import assert_allclose
+
+# import sys
+
+# import time as t
+
+import math
+
+import numpy as np
+
+import warnings
+
+# import scipy
+from scipy.sparse.linalg import lobpcg as scipy_lobpcg
+
+
+@pytest.mark.parametrize("dtype", [
+    cp.float32,
+    cp.float64
+])
+@pytest.mark.parametrize("k,n", [
+    (1, 10), (2, 10),
+    (1, 100), (2, 100), (5, 100),
+    (2, 1000), (10, 1000),
+    # (2,10000)
+])
+# TODO: debug and run tests for method ortho and basic
+#  currently they aren't that accurate
+@pytest.mark.parametrize("method", [
+    # "ortho",
+    # "basic",
+    'blopex'
+])
+@pytest.mark.parametrize("largest", [True,
+                                     False
+                                     ])
+@pytest.mark.parametrize("maxiter", [
+    20,
+    # 200, #Not practically required
+    # 2000
+])
+@pytest.mark.parametrize("isB", [True, False])
+def test_lobpcg(dtype, k, n, method, largest, maxiter, isB):
```

Comment:
Since the `blopex` implementation follows so closely to the Scipy implementation, it would be helpful to also include many of their tests directly: https://github.com/scipy/scipy/blob/v1.5.3/scipy/sparse/linalg/eigen/lobpcg/tests/test_lobpcg.py. This will also help verify that not only does the implementation provide the proper eigenvalues, but all the potential inputs that we support are handled properly.

Reply:
Thank you @cjnolet , those are some very elaborate tests that are definitely missing here! will add them 👍